File: leetcode/0003-longest-substring-without-repeating-characters/solution.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        if s=='':
            leng=0
        else:
            leng=1
        for i in range(0,len(s)):
            leng2=1
            l=[]
            for j in range(i+1,len(s)):
                if s[i]==s[j] or s[j] in l:
                    break
                elif s[i]!=s[j]:
                    leng2+=1
                    l.append(s[j])
                    if leng2>leng:
                        leng=leng2
        return leng
            
        """
        :type s: str
        :rtype: int
        """
        

File: leetcode/0003-longest-substring-without-repeating-characters/solution.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<int,int> chrmap;
        int left=0;
        int maxl=0;
        for(int i=0;i<s.size();i++){
            if(chrmap.find(s[i])!=chrmap.end() && chrmap[s[i]]>=left){
                left=chrmap[s[i]]+1;
            }
            chrmap[s[i]]=i;
            maxl=max(maxl,i-left+1);
        }
        return maxl;
    }
};
